### PR TITLE
Implement ExpandedDistribution

### PR DESCRIPTION
**Pull request by fehiepsi**
Addresses #554. The implementation is almost identical to Pyro version. I also made `enumerate_support(expand=True)` by default to match PyTorch behavior.

### CODE REVIEW

File: test/test_distributions.py [hunk 1]
```diff
@@ -982,3 +982,29 @@ def test_enumerate_support_smoke(jax_dist, params, support, batch_shape, expand)
...
+
+@pytest.mark.parametrize('jax_dist, sp_dist, params', CONTINUOUS + DISCRETE)
+@pytest.mark.parametrize('prepend_shape', [
+    (),
+    (2, 3),
+])
+@pytest.mark.parametrize('sample_shape', [
+    (),
+    (4,),
+])
```
Comment (neerajprad) [inline test/test_distributions.py:995]: I think we should also add tests to check the interaction of `.to_event` and `.expand` and `log_prob` (you could take a look at `check_sample_shapes` in `test_distributions` in Pyro which is used to test many of these corner cases), but feel free to do that in a separate PR since you are currently blocked by this. Also, scipy distributions don't consistently follow the tensor shaping semantics that we have in pytorch, so it might be best to not take that as a reference.

Comment (fehiepsi) [reply test/test_distributions.py:995]: Thanks, @neerajprad ! I'll add it to Masked PR.
